api/types/role.go

```diff
@@ -132,11 +137,38 @@ type Role interface {
 	GetWindowsLogins(RoleConditionType) []string
 	// SetWindowsLogins sets Windows desktop logins for allow or deny condition.
 	SetWindowsLogins(RoleConditionType, []string)
+
+	// GetSessionRequirePolicies returns the RBAC required policies for a session.
+	GetSessionRequirePolicies() []*SessionRequirePolicy
+	// SetSessionRequirePolicies sets the RBAC required policies for a session.
+	SetSessionRequirePolicies([]*SessionRequirePolicy)
+	// GetSessionJoinPolicies returns the RBAC join policies for a session.
+	GetSessionJoinPolicies() []*SessionJoinPolicy
+	// SetSessionJoinPolicies sets the RBAC join policies for a session.
+	SetSessionJoinPolicies([]*SessionJoinPolicy)
+	// GetSessionPolicySet returns the RBAC policy set for a role.
+	GetSessionPolicySet() SessionTrackerPolicySet
 }
 
 // NewRole constructs new standard role
-func NewRole(name string, spec RoleSpecV4) (Role, error) {
-	role := RoleV4{
+func NewRole(name string, spec RoleSpecV5) (Role, error) {
+	role := RoleV5{
+		Version: V3,
```

Comment:
Unfortunately, it is. Teleport has a large reliance on V3 role for a dozen systems and tests internally. We rely on V3 RBAC defaults which are laxer than V4/V5. Getting a way from V3 at some point would be good but it is a large and brittle change and not in scope for this PR:

Reply:
I added that function since new tests etc should probably rely on default V5 RBAC semantics when possible so we don't pile up technical debt as we add more tests relying on roles etc.

Reply:
That makes sense. Can you just add godocs to `NewRole` and `NewRoleV5` mentioning that the v5 version should be preferred for new code and that the older version remains for compatibility with old tests?